Sources/Sentry/Profiling/SentryLaunchProfiling.m

```diff
@@ -135,28 +117,33 @@
 
         SENTRY_LOG_INFO(@"Starting app launch profile at %llu", appLaunchSystemTime);
 
-        // don't worry about synchronizing the write here, as there should be no other tracing
-        // activity going on this early in the process. this codepath is also behind a dispatch_once
-        isTracingAppLaunch = [SentryProfiler startWithTracer:appLaunchTraceId];
+        SentryTransactionContext *context =
+            [[SentryTransactionContext alloc] initWithName:@"launch" operation:@"app.lifecycle"];
+        SentryTracerConfiguration *config = [SentryTracerConfiguration defaultConfiguration];
+        NSDictionary<NSString *, NSNumber *> *rates = appLaunchProfileConfiguration();
+        NSNumber *profilesRate = rates[kSentryLaunchProfileConfigKeyProfilesSampleRate];
+        NSNumber *tracesRate = rates[kSentryLaunchProfileConfigKeyTracesSampleRate];
+        if (profilesRate != nil && tracesRate != nil) {
+            config.profilesSamplerDecision =
+                [[SentrySamplerDecision alloc] initWithDecision:kSentrySampleDecisionYes
+                                                  forSampleRate:profilesRate];
+            context.sampleRate = tracesRate;
+        }
+        launchTracer = [[SentryTracer alloc] initWithTransactionContext:context
+                                                                    hub:nil
```

Comment:
I just noticed that this won't work, because on this branch we never assign a hub to the `launchTracer`, but that's fixed with https://github.com/getsentry/sentry-cocoa/pull/3634 💪